src/v/kafka/server/usage_aggregator.cc

```diff
@@ -109,16 +109,34 @@ usage& usage::operator+=(const usage& other) {
     return *this;
 }
 
-template<typename time_point>
-auto epoch_time_secs(time_point now) {
+/// The type that will be providing timestamps in form of seconds since
+/// epoch. ss::lowres_clock has no concept of epoch time, therefore it isn't
+/// used for generating timestampsp
+template<typename clock_type>
+using ts_type = std::conditional_t<
+  std::is_same_v<clock_type, ss::lowres_clock>,
+  ss::lowres_system_clock,
+  clock_type>;
+
+template<typename clock_type, typename duration>
+concept supports_epoch_time = requires(
+  std::chrono::time_point<clock_type, duration> t) { t.time_since_epoch(); };
```

Comment:
i don't understand this concept in that i think it will always be true, since iiuc all std:chrono::time_point supports time_since_epoch for all clock types.

Reply:
It's only supported for lowres_system_clock and the manual clock , not lowres_clock

Reply:
i still don't understand. i think this concept requirement is always satisfied.

Reply:
Yup this is the case, removing